### DISCUSSION TITLE
文章聚合

### DISCUSSION THREAD

**Discussion by jiangyangcreate** [travellings-link/travellings#1750] [发展建议]
聚合开往成员中优秀的文章-站点提供单独的曝光渠道，需要使用nodejs完成

效果类似：bilibili每周必看

维护者推荐制

**Comment by linlinzzo** [comment]
貌似有现成的程序 FreshRSS？

**Comment by linlinzzo** [comment]
貌似可以先通过Fresh RSS获取开往成员网站文章，然后像 saveweb 那样弄个静态聚合？（ https://github.com/saveweb/review-2022 ）

**Comment by jiangyangcreate** [comment]
成品效果应该没有歧义。实现方法不限，可以直接做出成品。那这个issue就交付给你了。

**Comment by linlinzzo** [comment]
> 成品效果应该没有歧义。实现方法不限，可以直接做出成品。那这个issue就交付给你了。

麻烦帮忙装一下FreshRSS。（貌似之后服务越来越多了）@Luochancy

**Comment by jiangyangcreate** [comment]
先在测试环境下(本地)运行，确认内存和CPU峰值占用，确认后再部署，避免服务崩。另外10月的排期10月做。

**Comment by Luochancy** [comment]
抱歉我最近才看到issue（没咋看邮箱😭）
目前打算基于CentOS 7-2009搭建本地测试环境，目前我手上能用到的测试环境的配置是E5-2680V3。
得看看试跑到啥时候才可以🤔

**Comment by jiangyangcreate** [comment]
如果可以的话先出一份框架选择调研：
受欢迎度
功能分析
语言
易集成：是否提供纯API易于集成到其他框架
易搭建
负载均衡：2核2G做压测
响应速度

为什么不是 huginn （更受欢迎） 、RSSHub（更契合当前技术栈）等等.....

可以选择FreshRSS，但是需要一个调研表，证明它确实是最好的选择。 @linlinzzo @Luochancy 

这一步会在开始的时候额外的花费时间，但是可以避免了后续迁移、技术协作的隐藏时间。

**Comment by Luochancy** [comment]
目前我用4C4G做的测试环境
通过Docker方式部署的FreshRSS，支持多语言。htop显示整机（包括CentOS7-2009）内存占用500MB左右，而CPU基本上就是没在动。
研究了下FreshRSS的API，应该是够用（具体看https://freshrss.github.io/FreshRSS/en/developers/06_Fever_API.html）

我过几天会再开个小鸡作为另一个测试环境测测其他的RSS系统，需要点时间
在做完性能测试以后我再给出一个调研表

倒是我不清楚怎么压测这种RSS系统😭还请各位老指点下@jiangmiemie

**Comment by Luochancy** [comment]
后面4C4G的配置会降到2C2G做持续性的性能监控，目前4C4G仅仅是我服务器资源有点膨胀随便给的（

**Comment by jiangyangcreate** [comment]
RSS一般不直接用，都是调用接口。

接口压测一般是确定请求数量，选择合适的请求工具。

数量上：我们1k的成员站可以以1000为基本单位开始测试，每秒1000并发、2000并发以此类推，看下在第几轮挂掉。
工具上：选一些开源在线的接口测试工具就行，自动出报告。一般要输入请求地址、请求类型与数据等等。

最后应该会获得一个报告：图表形式或者表格形式：随着请求变多资源耗尽，然后可以查看它资源消耗是指数型的增长还是普通线性增长。

**Comment by Luochancy** [comment]
okok
感谢大佬指点~
